ethereum/rpc/namespaces/eth/api.go

```diff
@@ -529,6 +534,20 @@ func (e *PublicAPI) Call(args evmtypes.CallArgs, blockNr rpctypes.BlockNumber, _
 	return (hexutil.Bytes)(data.Ret), nil
 }
 
+// isIntrinsicGasError try to reflex the error internal, to decide if it's a `core.ErrIntrinsicGas`,
+// pretty hacky
+func isIntrinsicGasError(err error) bool {
+	if se, ok := status.FromError(err); ok {
+		wrapper := status.Error(
+			codes.Internal,
+			core.ErrIntrinsicGas.Error(),
+		)
+		sdkWrapper := sdkerrors.Wrap(sdkerrors.ErrUnknownRequest, wrapper.Error())
+		return se.Message() == sdkWrapper.Error()
```

Comment:
can you use `errors.Is`?

Reply:
No, it’s simply a string after transferred to client.

Reply:
what about `strings.Contains`

Reply:
this is out dated now, current version is better